raptorWeb/authprofiles/views.py

```diff
@@ -0,0 +1,347 @@
+
+        else:
+
+            dictionary["register_form"] = register_form
+            dictionary["extra_form"] = extra_form
+
+            return render(request, self.template_name, context=dictionary)
+
+class User_Login_Form(TemplateView):
+    """
+    Returns a form for a user to login with Username and Password
+    """
+    login_form = UserLoginForm()
+
+    def get(self, request):
+
+        if request.headers.get('HX-Request') == "true":
+            dictionary = player_poller.currentPlayers_DB
+            dictionary["login_form"] = self.login_form
+            template_name = join(settings.AUTH_TEMPLATE_DIR, 'login.html')
+            return render(request, template_name, context=dictionary)
+        else:
+            return HttpResponseRedirect('../')
+
+    def post(self, request):
+
+        login_form = UserLoginForm(request.POST)
+        dictionary = player_poller.currentPlayers_DB
+        dictionary["login_form"] = self.login_form
+
+        if login_form.is_valid():
+            username = login_form.cleaned_data["username"]
+            password = login_form.cleaned_data["password"]
+            user = authenticate(username=username, password=password)
+            if user:
+                LOGGER.info("User logged in!")
+                login(request, user)
+                return HttpResponseRedirect(self.request.META.get('HTTP_REFERER'))
+            else:
+                return HttpResponse("Account does not exist")
+        else:
+            messages.error(request, login_form.errors.as_text().replace('* __all__', ''))
+            return HttpResponseRedirect(self.request.META.get('HTTP_REFERER'))
+
+class UserLogin_OAuth(TemplateView):
+
+    def get(self, request):
+        return redirect(settings.DISCORD_AUTH_URL)
+
+class UserLogin_OAuth_Success(TemplateView):
+
+    def get(self, request):
+
+        try:
+            discord_code = request.GET.get('code')
+            user_info = discordAuth.exchange_code(discord_code)
+            discord_user = authenticate(request, user=user_info)
+            try:
+                login(request, discord_user, backend='authprofiles.auth.DiscordAuthBackend')
+            except AttributeError:
+                login(request, list(discord_user).pop(), backend='authprofiles.auth.DiscordAuthBackend')
+            return redirect('../../')
+        except KeyError:
+            return HttpResponseRedirect("../login")
+
+@login_required
+def user_logout(request):
+    """
+    Log out the signed in user
+    """
+    logout(request)
+    LOGGER.info("User logged out!")
+    return HttpResponseRedirect(request.META.get('HTTP_REFERER'))
+
+class All_User_Profile(TemplateView):
+    """
+    Displays all User Profiles
+    """
+    template_name = join(settings.AUTH_TEMPLATE_DIR, 'all_profiles.html')
+
+    def get(self, request):
+        instance_dict = player_poller.currentPlayers_DB
+        instance_dict = viewContext.update_context(context = instance_dict)
```

Comment:
As mentioned in Line 34 review, the `viewContext` module needs to not be used here. Information gathered from there must be gathered here instead.